scripts/zkai_native_seq32_attention_mlp_predecommit_opening_policy_gate.py

```
@@ -0,0 +1,593 @@
+
+
+def validate_payload(payload: dict[str, Any]) -> None:
+    item = copy.deepcopy(payload)
+    supplied_commitment = item.pop("payload_commitment", None)
+    mutation_result = item.pop("mutation_result", None)
+    validate_base_payload(item)
+    if supplied_commitment != payload_commitment(payload):
+        raise PredecommitOpeningPolicyGateError("payload commitment drift")
+    expected_mutation = run_mutations(item)
+    if mutation_result != expected_mutation:
+        raise PredecommitOpeningPolicyGateError("mutation result drift")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)